TaskLayer/SearchTask/SearchTask.cs

```diff
@@ -744,6 +744,170 @@ public static void WriteMzidentml(IEnumerable<PeptideSpectralMatch> items, List<
+                 {
+                     name = CommonParameters.DigestionParams.Protease.Name,
+                     specificity = new pepXML.Generated.msms_pipeline_analysisMsms_run_summarySample_enzymeSpecificity[1]
+                     {
+                         new pepXML.Generated.msms_pipeline_analysisMsms_run_summarySample_enzymeSpecificity
+                         {
+                             cut = proteaseC,
+                             no_cut = proteaseNC,
+                         }
+                     }
+                 },
+
+                 search_summary = new pepXML.Generated.msms_pipeline_analysisMsms_run_summarySearch_summary[1]
+                 {
+                     new pepXML.Generated.msms_pipeline_analysisMsms_run_summarySearch_summary
+                     {
+                         base_name = filePathNoExtension,
+                         //Contact for support
+                         //search_engine = pepXML.Generated.engineType.MetaMorpheus 
+                         search_engine_version = GlobalVariables.MetaMorpheusVersion,
+                         precursor_mass_type = pepXML.Generated.massType.monoisotopic,
+                         fragment_mass_type = pepXML.Generated.massType.monoisotopic,
+                         search_id = 1,
+                         //generate database information
+                         search_database = new pepXML.Generated.msms_pipeline_analysisMsms_run_summarySearch_summarySearch_database
+                         {
+                             local_path = database.First().FilePath,
+                             type = pepXML.Generated.msms_pipeline_analysisMsms_run_summarySearch_summarySearch_databaseType.AA,
+                         },
+                         enzymatic_search_constraint = new pepXML.Generated.msms_pipeline_analysisMsms_run_summarySearch_summaryEnzymatic_search_constraint
+                         {
+                             enzyme = CommonParameters.DigestionParams.Protease.Name,
+                             max_num_internal_cleavages = CommonParameters.DigestionParams.MaxMissedCleavages.ToString(),
+                         },
+
+                         parameter = para.ToArray()
+                     }
+                 },
+                 }
+             };
+
+            _pepxml.msms_run_summary[0].spectrum_query = new pepXML.Generated.msms_pipeline_analysisMsms_run_summarySpectrum_query[items.Count];
+
+            var searchHits = new List<pepXML.Generated.msms_pipeline_analysisMsms_run_summarySpectrum_querySearch_resultSearch_hit>();
+            for (int i = 0; i < items.Count; i++)
+            {
+                var pepWithSetMods = items[i].CompactPeptides.First().Value.Item2.First();
+                int pepXmlMods = pepWithSetMods.allModsOneIsNterminus.Count;
+                var mods = new List<pepXML.Generated.modInfoDataTypeMod_aminoacid_mass>();
+                for (int j = 0; j < pepXmlMods; j++)
```

Comment:
this `For ` loop is pretty inefficient, it is generating a new `List` with every iteration which is not needed.

```
foreach(var mod in pepWithSetMods.allModsOneIsNterminus)
{
   var pepXmlMod = new pepXML.Generated.modInfoDataTypeMod_aminoacid_mass
   {
      mass = mod.Value.monoisotopicMass,
      position = mod.Key - 1
   };

   pepXmlMods.Add(pepXmlMod);
}
```